plugins/modules/nd_api_key.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Dev Sinha (@DevSinha13) <devsinh@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: nd_api_key
+version_added: "1.4.0"
+short_description: Manage API keys in Nexus Dashboard
+description:
+- Manage API keys in Cisco Nexus Dashboard (ND).
+- It supports creating, updating, querying, and deleting API keys.
+author:
+- Dev Sinha (@DevSinha13)
+options:
+  api_key_id:
+    description:
+    - The ID of the API key.
+    type: str
+  api_key_name:
+    description:
+    - The name of the API key.
+    - Required when state is 'present'.
+    - Must be 1-32 characters long.
```

Comment:
For description this is not correct, I can change it to empty string. Only on create this seems to be an issue in UI. For which we likely want to raise an issue towards the product